src/client/opamAdminCommand.ml

```diff
@@ -16,13 +16,33 @@ open Cmdliner
 
 type command = unit Cmdliner.Term.t * Cmdliner.Cmd.info
 
-let checked_repo_root () =
+let repo_version_lt repo_root v =
+  let v' =
+    let open OpamStd.Option.Op in
+    (OpamFile.Repo.read_opt
+       (OpamRepositoryPath.repo repo_root)
+     >>= OpamFile.Repo.opam_version)
+    +! (OpamVersion.of_string "1.2")
```

Comment:
I think this should also probably be be extracted from OpamAdminRepoUpgrade line 479, put at the top of that module same as upgradeto_version_string, exported and used here instead.